http-clients/src/main/java/com/palantir/remoting/http/Decorators.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.remoting.http;
+
+import com.google.common.base.Preconditions;
+import com.google.common.reflect.TypeToken;
+import java.lang.reflect.Constructor;
+import java.util.Arrays;
+import java.util.Set;
+
+/**
+ * Methods that allow classes that use the decorator pattern to be instantiated in a declarative manner.
+ */
+final class Decorators {
```

Comment:
OK cool, thanks for the feedback, totally fair. May take a look at this simplification when I have some spare cycles, but not urgent.

Hmm, went to GitColony but didn't see this PR there?